tasks/run_kops.yml

```diff
@@ -0,0 +1,178 @@
+      export AWS_PROFILE="{{ kops_aws_profile }}";
+      {% endif %}
+      # Don't fail to show that remote has nothing yet
+      kops get cluster -o yaml \
+        --state s3://{{ cluster.s3_bucket_name }} \
+        --name {{ cluster.name }} || true
+    source_type: string
+    target_type: command
+    diff: yaml
+    diff_yaml_ignore: ['creationTimestamp']
+  check_mode: False
+  register: _kops_diff_state_cluster
+  when:
+    - kops_update is defined
+    - kops_update == 'state' or kops_update == 'update' or kops_update == 'all'
+
+- name: diff generated instance group config against remote state store
+  udiff:
+    source: "{{ lookup('template', 'instance-groups.yml.j2') }}"
+    target: |
+      {% if kops_aws_profile %}
+      # If using boto_profile, make kops aware of it
+      export AWS_PROFILE="{{ kops_aws_profile }}";
+      {% endif %}
+      # Don't fail to show that remote has nothing yet
+      kops get instancegroups -o yaml \
+        --state s3://{{ cluster.s3_bucket_name }} \
+        --name {{ cluster.name }} || true
+    source_type: string
+    target_type: command
+    diff: yaml
+    diff_yaml_ignore: ['creationTimestamp']
+  check_mode: False
+  register: _kops_diff_state_instancegroups
+  when:
+    - kops_update is defined
+    - kops_update == 'state' or kops_update == 'update' or kops_update == 'all'
+
+- name: diff ssh public key against remote state store
+  udiff:
+    source: |
+      # Test SSH key for validity
+      if ! bash -c "ssh-keygen -lf <(echo '{{ lookup('template', 'ssh-key.pub.j2') }}')" >/dev/null; then
+        exit 1;
+      fi
+      # Get SSH key fingerprint only
+      bash -c "ssh-keygen -lE md5 -f <(echo '{{ lookup('template', 'ssh-key.pub.j2') }}')" \
+        | awk '{ print $2}' | sed 's/^MD5://g'
+    target: |
+      {% if kops_aws_profile %}
+      # If using boto_profile, make kops aware of it
+      export AWS_PROFILE="{{ kops_aws_profile }}";
+      {% endif %}
+      # Don't fail to show that remote has nothing yet
+      kops get secret admin --type SSHPublicKey -o table \
+        --state s3://{{ cluster.s3_bucket_name }} \
+        --name {{ cluster.name }} \
+        | grep -Eo "[:a-f0-9]{16,}" || true
+    source_type: command
+    target_type: command
+  check_mode: False
+  register: _kops_diff_state_sshkey
+  when:
+    - kops_update is defined
+    - kops_update == 'state' or kops_update == 'update' or kops_update == 'all'
+
+
+###
+### Update State Store
+###
+- name: update state store for cluster spec
+  shell: |
+    {% if kops_aws_profile %}
+    # If using boto_profile, make kops aware of it
+    export AWS_PROFILE="{{ kops_aws_profile }}";
+    {% endif %}
+    kops replace -v 9 --force \
+      --state s3://{{ cluster.s3_bucket_name }} \
+      --name {{ cluster.name }} \
+      -f <(echo "{{ lookup('template', 'cluster.yml.j2') }}")
+  args:
+    executable: bash
+  when:
+    - _kops_diff_state_cluster['changed']
```

Comment:
IIRC, the pipe notation is favourable over square brackets and the dot notation.